pyomo/contrib/mindtpy/util.py

```diff
@@ -844,3 +849,65 @@ def set_up_logger(config):
     ch.setFormatter(formatter)
     # add the handlers to logger
     config.logger.addHandler(ch)
+
+
+def get_dual_integral(solve_data):
+    dual_integral = 0
+    dual_bound_progress = solve_data.dual_bound_progress.copy()
+    # Initial dual bound is set to inf or -inf. To calculate dual integral, we set
+    # initial_dual_bound to 10% greater or smaller than the first_found_dual_bound.
+    # TODO: check if the calculation of initial_dual_bound needs to be modified.
+    for dual_bound in dual_bound_progress:
+        if dual_bound != dual_bound_progress[0]:
+            break
+    for i in range(len(dual_bound_progress)):
```

Comment:
`for i, dual_bound in enumerate(dual_bound_progress)` maybe?

Reply:
If we want to change the value of the elements in a list, we need to use `dual_bound_progress[i]` .

Reply:
Alright!